python_modules/dagster/dagster/_core/definitions/declarative_scheduling/operators/since_operator.py

```diff
@@ -0,0 +1,49 @@
+from typing import Sequence
+
+from dagster._serdes.serdes import whitelist_for_serdes
+
+from ..scheduling_condition import SchedulingCondition, SchedulingResult
+from ..scheduling_context import SchedulingContext
+
+
+@whitelist_for_serdes
+class SinceCondition(SchedulingCondition):
+    primary_condition: SchedulingCondition
+    reference_condition: SchedulingCondition
```

Comment:
How about _trigger_ and _reset_ condition instead of _primary_ and _reference_?":

"During any tick where the reset condition becomes true, the since condition state resets to False. It will remain false until we encounter a tick where the trigger condition becomes true, at which point the since condition becomes stays true. It remains true until the reset condition is true again."


![image.png](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/wS38mdcdU6aAeJobBdry/17917bc3-7501-4700-ae1a-cb69c23d3351.png)